auth/src/main/java/com/vimeo/networking2/internal/AuthenticatorImpl.kt

```diff
@@ -58,12 +58,22 @@ internal class AuthenticatorImpl(
         token: String,
         email: String,
         marketingOptIn: Boolean,
-        authenticationErrorMessage: String,
+        socialAuthType: SocialAuthType,
         authCallback: AuthCallback
     ): VimeoRequest {
 
+        val tokenField = when(socialAuthType) {
```

Comment:
This code could be simplified if instead of passing an `enum` into this function, we passed a `data class` that encapsulated the dynamic properties. e.g.
```kotlin
data class SocialAuthRequestParams(
  val tokenFieldName: String,
  val errorMessage: String
)
```
Or these parameters could be added to the `socialAuthenticate` declaration itself.